pymc/backends/arviz.py

```diff
@@ -165,6 +165,17 @@ def __init__(
                         UserWarning,
                     )
             self.ntune = len(self.trace) - self.ndraws
+
+            for strace in self.trace._straces.values():
+                if strace._stats is not None:
+                    for stat in strace._stats:
+                        for k, v in stat.items():
+                            if isinstance(v, np.ndarray) and isinstance(v[0], np.ndarray):
+                                if self.nchains > 1:
+                                    stat[k] = np.vstack(v)
```

Comment:
Here I have a hard time understanding what the code does.
Mostly because the things are being modified inplace.

**Over which dimension are you stacking things and why?** Isn't this just replacing arrays by vstacked versions of themself? I mean the line that's not test-covered.

Reply:
Currently if the stat is an array, we get and array of arrays, instead of an array of the correct shape. Thus the solution proposed here is to stack the array of arrays.

Happy to fix this with a different approach, but not sure what that approach should be.

Reply:
Probably something here: https://github.com/pymc-devs/pymc/blob/main/pymc/backends/base.py#L204-L206

This code relies on the standard `BaseTrace` API and in that retrieval mechanism it should be okay to attempt a stacking & fall back to an array of arrays if the stacking doesn't work.